libcontainer/nsenter/nsexec.c

```diff
@@ -192,22 +194,74 @@ static void update_setgroups(int pid, enum policy_t setgroup)
 	}
 }
 
-static void update_uidmap(int pid, char *map, size_t map_len)
+static int attempt_mapping_tool(const char *app, int pid, char *map, size_t map_len)
+{
+#define MAX_ARGV 20
+	char *argv[MAX_ARGV];
+	char pid_fmt[16];
+	int argc = 0;
+	size_t i;
+	int child;
+
+	sprintf (pid_fmt, "%d", pid);
```

Comment:
AFAICS #1403 is doing a very similar thing for preparing the argv for `newuidmap/newgidmap`

The main difference is that here I am not using `strtok`  and parsing the string inline without allocating any memory.  I've avoided input validation here as it would be superfluous since newuidmap/newgidmap do that already.